README.md

```diff
@@ -1,49 +1,219 @@
+Before getting started, review important information about this feature:
+
+- [Terminology](#terminology)
+- [Getting help and providing feedback](#getting-help-and-providing-feedback)
+
+Build, deploy, and operate your connector:
+
+- [Building a connector](#building-a-connector)
+- [Operating the connector](#operating-the-connector)
+- [Moving to production](#moving-to-production)
+
+How to publish your connector:
+
+- [Contribute to the repository](#contribute-to-the-repository-)
+
+Reference:
+
+- [Connector protocol reference](docs/CONNECTOR_PROTOCOL.md)
+
+## Terminology
+
+- `connector client` - specific light-weight connector implementation, open-code. Connector clients can be built by Elastic or community built.
+- `connector service` - the app that runs the asynchronous loop that calls Elasticsearch on a regular basis to check whether syncs need to happen.
+- `connector framework` - framework and libraries used to customize and build connector clients. Basically, it's what this repository is.
+- `connector packages` - a previous version of the connector clients. Refer to the [8.3 branch](https://github.com/elastic/connectors-ruby/tree/8.3) if you're looking for connector packages. Also, read more about them in the [custom connector packages guide](https://www.elastic.co/guide/en/workplace-search/8.4/custom-connector-package.html).
+- `data source` - file/database/service that provides data to be ingested into Elasticsearch. 
+
+## Getting help and providing feedback
+
+May want to duplicate the content I put in the Elastic doc: https://github.com/elastic/enterprise-search-pubs/pull/2631
+
+## Building a connector
+
+This section has its own structure, so mini TOC to provide an overview:
+
+- [Implementing the connector protocol](#implementing-the-connector-protocol)
+- [Using the connector framework](#using-the-connector-framework)
+- [Operating the connector](#operating-the-connector)
+- [Testing the integration](#testing-the-integration)
+
+
+### Implementing the connector protocol
+
+The [connector protocol](docs/CONNECTOR_PROTOCOL.md) is document-based and building a connector is language- and tool-agnostic.
+To create an Elastic connector, you need to implement this protocol. At a high level, you need to create an application (the connector service) that can read and write to a specific document in Elasticsearch that represents the connector. This document "registers" the connector with Kibana, keeps the configuration that is made via the Kibana UI, as well as the scheduling configuration and the state information for the service. You need to to update the connector state to keep up with the current status of the connector application and of the third-party data source. You also need to log sync jobs to an additional index, so that the history of synchronization tasks would be available. And of course, you need to write the results of the synchronization to the Elasticsearch document index, created for this connector service.
```

Comment:
```suggestion
To create an Elastic connector, you need to implement this protocol. At a high level, you need to create an application (the connector service) that can read and write to a specific document in Elasticsearch that represents the connector. This document "registers" the connector with Kibana, keeps the configuration that is made via the Kibana UI, as well as the scheduling configuration and the state information for the service. You need to update the connector state to keep up with the current status of the connector application and of the third-party data source. You also need to log sync jobs to an additional index, so that the history of synchronization tasks would be available. And of course, you need to write the results of the synchronization to the Elasticsearch document index, created for this connector service.
```